eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentLifecycle.scala

```diff
@@ -219,6 +219,10 @@ object PaymentLifecycle {
   case class UnreadableRemoteFailure(route: Seq[Hop]) extends PaymentFailure
   case class PaymentFailed(id: UUID, paymentHash: ByteVector32, failures: Seq[PaymentFailure]) extends PaymentResult
 
+  sealed trait PaymentOptions
```

Comment:
The reason I introduced that is because it's useful for AMP and Trampoline.
This allows changing the `case object TlvPayload` to a `sealed trait` instead with methods that allow setting additional tlvs inside the hops.